mcli/mcli-hf-eval.yaml

```diff
@@ -41,250 +41,14 @@ parameters:
-  #     name: hf_causal_lm
-  #     pretrained_model_name_or_path: togethercomputer/RedPajama-INCITE-7B-Base
-  #     init_device: cpu
-  #     pretrained: true
-  #     use_auth_token: false
-  # -
-  #   model_name: huggyllama/llama-13b
-  #   # Tokenizer
-  #   tokenizer:
-  #     name: huggyllama/llama-13b
-  #     kwargs:
-  #       model_max_length: ${max_seq_len}
-
-  #   model:
-  #     name: hf_causal_lm
-  #     pretrained_model_name_or_path: huggyllama/llama-13b
-  #     init_device: cpu
-  #     pretrained: true
-  #     use_auth_token: false
-  # -
-  #   model_name: huggyllama/llama-7b
-  #   # Tokenizer
-  #   tokenizer:
-  #     name: huggyllama/llama-7b
-  #     kwargs:
-  #       model_max_length: ${max_seq_len}
-
-  #   model:
-  #     name: hf_causal_lm
-  #     pretrained_model_name_or_path: huggyllama/llama-7b
-  #     init_device: cpu
-  #     pretrained: true
-  #     use_auth_token: false
-  # -
-  #   model_name: EleutherAI/pythia-12b
-  #   # Tokenizer
-  #   tokenizer:
-  #     name: EleutherAI/pythia-12b
-  #     kwargs:
-  #       model_max_length: ${max_seq_len}
-
-  #   model:
-  #     name: hf_causal_lm
-  #     pretrained_model_name_or_path: EleutherAI/pythia-12b
-  #     init_device: cpu
-  #     pretrained: true
-  #     use_auth_token: false
-  # -
-  #   model_name: EleutherAI/pythia-6.9b
-  #   # Tokenizer
-  #   tokenizer:
-  #     name: EleutherAI/pythia-6.9b
-  #     kwargs:
-  #       model_max_length: ${max_seq_len}
-
-  #   model:
-  #     name: hf_causal_lm
-  #     pretrained_model_name_or_path: EleutherAI/pythia-6.9b
-  #     init_device: cpu
-  #     pretrained: true
-  #     use_auth_token: false
-  # -
-  #   model_name: facebook/opt-6.7b
-  #   # Tokenizer
-  #   tokenizer:
-  #     name: facebook/opt-6.7b
-  #     kwargs:
-  #       model_max_length: ${max_seq_len}
-
-  #   model:
-  #     name: hf_causal_lm
-  #     pretrained_model_name_or_path: facebook/opt-6.7b
-  #     init_device: cpu
-  #     pretrained: true
-  #     use_auth_token: false
-  # -
-  #   model_name: stabilityai/stablelm-tuned-alpha-7b
-  #   # Tokenizer
-  #   tokenizer:
-  #     name: stabilityai/stablelm-tuned-alpha-7b
-  #     kwargs:
-  #       model_max_length: ${max_seq_len}
-
-  #   model:
-  #     name: hf_causal_lm
-  #     pretrained_model_name_or_path: stabilityai/stablelm-tuned-alpha-7b
-  #     init_device: cpu
-  #     pretrained: true
-  #     use_auth_token: false
-  # -
-  #   model_name: EleutherAI/gpt-j-6b
-  #   # Tokenizer
-  #   tokenizer:
-  #     name: EleutherAI/gpt-j-6b
-  #     kwargs:
-  #       model_max_length: ${max_seq_len}
-
-  #   model:
-  #     name: hf_causal_lm
-  #     pretrained_model_name_or_path: EleutherAI/gpt-j-6b
-  #     init_device: cpu
-  #     pretrained: true
-  #     use_auth_token: false
-
 
   # FSDP config for model sharding
   fsdp_config:
     sharding_strategy: FULL_SHARD
     mixed_precision: FULL
+    forward_prefetch: True
+    limit_all_gathers: True
```

Comment:
why are these needed?

Reply:
Brandon found that it made FSDP + inference faster to use these flags